src/StudentsController.ts

```diff
@@ -165,10 +169,53 @@ export class StudentsController {
         await this.#studentsCollection.update(oldDoc, student.toJSON());
     }
 
+    public async pseudonymizeStudent(student: Student): Promise<Student> {
+        // Only keep the id of the student
+        const oldDoc = await this.#studentsCollection.read(student.id.toString());
+        const pseudonymizedDoc = {
+            id: oldDoc.id
+        };
+
+        await this.#studentsCollection.update(oldDoc, pseudonymizedDoc);
+
+        return Student.from(pseudonymizedDoc);
+    }
+
     public async deleteStudent(student: Student): Promise<void> {
+        if (student.correspondingRelationshipId) {
+            let relationship = await this.getRelationship(student.correspondingRelationshipId);
+
+            if (relationship.status === RelationshipStatus.Pending) {
+                await this.services.transportServices.relationships.rejectRelationship({ relationshipId: student.correspondingRelationshipId.toString() });
+            } else if (relationship.status === RelationshipStatus.Active) {
+                await this.services.transportServices.relationships.terminateRelationship({ relationshipId: student.correspondingRelationshipId.toString() });
+            }
+
+            relationship = await this.getRelationship(student.correspondingRelationshipId);
+
+            if (
+                relationship.status === RelationshipStatus.Terminated ||
+                relationship.status === RelationshipStatus.Rejected ||
+                relationship.status === RelationshipStatus.Revoked ||
+                relationship.status === RelationshipStatus.DeletionProposed
+            ) {
+                await this.services.transportServices.relationships.decomposeRelationship({ relationshipId: student.correspondingRelationshipId.toString() });
+            }
+        }
+
+        if (student.correspondingRelationshipId) {
+            await this.services.transportServices.relationshipTemplates.deleteRelationshipTemplate({ templateId: student.correspondingRelationshipId.toString() });
+        }
```

Comment:
that should already happen under the hood (decomposition with a template with only one allocation)